src/beast/beast/utility/impl/PropertyStream.cpp

```diff
@@ -192,37 +193,31 @@ std::string const& PropertyStream::Source::name () const
 
 void PropertyStream::Source::add (Source& source)
 {
-    SharedState::Access state (m_state);
-    SharedState::Access childState (source.m_state);
-    bassert (childState->parent == nullptr);
-    state->children.push_back (childState->item);
-    childState->parent = this;
+    std::lock_guard<std::recursive_mutex> _l(lock_);
+    std::lock_guard<std::recursive_mutex> _cl(source.lock_);
```

Comment:
When locking multiple locks like this, consider `std::lock (lock_, source.lock_);
std::lock_guard l(lock_, std::adopt_lock);
std::lock_guard cl(source.lock_, std::adopt_lock);`

See: http://en.cppreference.com/w/cpp/thread/lock